scripts/convert_vq_diffusion_to_diffusers.py

```diff
@@ -0,0 +1,508 @@
+            f"{diffusers_resnet_prefix}.conv_shortcut.bias": checkpoint[f"{resnet_prefix}.nin_shortcut.bias"],
+        })
+
+    return rv
+
+def attention_to_diffusers_checkpoint(checkpoint, *, diffusers_attention_prefix, attention_prefix):
+    return {
+        # group_norm
+        f"{diffusers_attention_prefix}.group_norm.weight": checkpoint[f"{attention_prefix}.norm.weight"],
+        f"{diffusers_attention_prefix}.group_norm.bias": checkpoint[f"{attention_prefix}.norm.bias"],
+
+        # query
+        f"{diffusers_attention_prefix}.query.weight": checkpoint[f"{attention_prefix}.q.weight"],
+        f"{diffusers_attention_prefix}.query.bias": checkpoint[f"{attention_prefix}.q.bias"],
+
+        # key
+        f"{diffusers_attention_prefix}.key.weight": checkpoint[f"{attention_prefix}.k.weight"],
+        f"{diffusers_attention_prefix}.key.bias": checkpoint[f"{attention_prefix}.k.bias"],
+
+        # value
+        f"{diffusers_attention_prefix}.value.weight": checkpoint[f"{attention_prefix}.v.weight"],
+        f"{diffusers_attention_prefix}.value.bias": checkpoint[f"{attention_prefix}.v.bias"],
+
+        # proj_attn
+        f"{diffusers_attention_prefix}.proj_attn.weight": checkpoint[f"{attention_prefix}.proj_out.weight"],
+        f"{diffusers_attention_prefix}.proj_attn.bias": checkpoint[f"{attention_prefix}.proj_out.bias"],
+    }
+
+###############################################
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--only-vqvae",
+        action="store_true",
+        help="Set flag to only convert the VQVAE model. checkpoint_path and original_config_file must point to checkpoints and configs only for the VQVAE, not the whole diffusion model. Note that the entirety of the diffusers VQDiffusionPipeline will be saved."
+    )
+
+    parser.add_argument(
+        "--checkpoint_path", 
+        default=None, 
+        type=str, 
+        required=True, 
+        help="Path to the checkpoint to convert."
+    )
+
+    parser.add_argument(
+        "--original_config_file",
+        default=None,
+        type=str,
+        required=True,
+        help="The YAML config file corresponding to the original architecture.",
+    )
+
+    parser.add_argument(
+        "--dump_path", 
+        default=None, 
+        type=str, 
+        required=True, 
+        help="Path to the output model."
+    )
+
+    args = parser.parse_args()
+
+    # The yaml file contains annotations that certain values should
+    # loaded as tuples. By default, OmegaConf will panic when reading
+    # these. Instead, we can manually read the yaml with the FullLoader and then
+    # construct the OmegaConf object.
+    with open(args.original_config_file) as f:
+        original_config = yaml.load(f, FullLoader)
+
+    original_config = OmegaConf.create(original_config).model
+
+    checkpoint = torch.load(args.checkpoint_path)['model']
```

Comment:
nit: by default, torch loads model from CUDA. add a flag to specify device-type?

Reply:
sg! added on the transformer branch